boa/src/builtins/map/mod.rs

```diff
@@ -309,6 +329,20 @@ impl Map {
     ) -> JsResult<JsValue> {
         let key = args.get_or_undefined(0);
 
+        let temp_v;
+
+        let key = match key {
+            JsValue::Rational(r) => {
+                if r.is_zero() {
+                    temp_v = JsValue::Rational(0f64);
+                    &temp_v
+                } else {
+                    key
+                }
+            }
+            _ => key,
+        };
+
```

Comment:
```suggestion
        const JS_ZERO: &JsValue = &JsValue::Rational(0f64);

        let key = args.get_or_undefined(0);

        let key = match key {
            JsValue::Rational(r) => {
                if r.is_zero() {
                    JS_ZERO
                } else {
                    key
                }
            }
            _ => key,
        };
```

A constant is more readable than a free `temp_v`.